src/main/java/com/google/cloud/spanner/jdbc/JdbcTypeConverter.java

```diff
@@ -183,9 +190,39 @@ static Object convert(Object value, Type type, Class<?> targetType) throws SQLEx
               ZoneId.systemDefault());
         }
       }
+      if (AbstractMessage.class.isAssignableFrom(targetType)) {
+        if (type.getCode() == Code.PROTO || type.getCode() == Code.BYTES) {
+          Method parseMethodParseFrom = targetType.getMethod("parseFrom", byte[].class);
+          return targetType.cast(parseMethodParseFrom.invoke(null, value));
+        }
+      }
+      if (ProtocolMessageEnum.class.isAssignableFrom(targetType)) {
+        if (type.getCode() == Code.ENUM || type.getCode() == Code.INT64) {
+          Method parseMethodForNumber = targetType.getMethod("forNumber", int.class);
+          return targetType.cast(
+              parseMethodForNumber.invoke(
+                  null, AbstractJdbcWrapper.checkedCastToInt((Long) value)));
+        }
+      }
       if (targetType.equals(java.sql.Array.class)) {
         if (type.getCode() == Code.ARRAY) return value;
       }
+      if (targetType.isArray() && type.getCode() == Code.ARRAY) {
+        if (type.getArrayElementType().getCode() == Code.PROTO
+            || type.getArrayElementType().getCode() == Code.BYTES) {
```

Comment:
I worry a bit about this last part (`type.getArrayElementType().getCode() == Code.BYTES`). Won't that potentially try to convert a regular BYTES array into an array of PROTO messages?

Reply:
@olavloite 
This part of code gets called when using `getObject()` method of `JDBCResultSet` class which takes a target class type `Class<T>` as input argument.
1. In this case the regular BYTES array gets converted to an array of proto message only when the target array type is a proto message class. This check happens which in this if condition.
2. In case when the target type is BYTES array then it comes out of this if loop and goes to the loop that satisfies this target type. Currently when we pass a BYTES array it throws an `Cannot convert` exception as these array types are not handled. We only allow Array.class as target type.